zcash_client_sqlite/src/wallet/sapling.rs

```diff
@@ -132,14 +132,34 @@ pub(crate) fn get_spendable_sapling_notes(
     anchor_height: BlockHeight,
     exclude: &[ReceivedNoteId],
 ) -> Result<Vec<ReceivedSaplingNote<ReceivedNoteId>>, SqliteClientError> {
+    let mut stmt_unscanned_tip = conn.prepare_cached(
+        "SELECT 1 FROM v_sapling_shard_unscanned_ranges unscanned
+         WHERE unscanned.shard_index IN (SELECT MAX(shard_index) FROM sapling_tree_shards)
+         AND unscanned.subtree_start_height <= :anchor_height",
```

Comment:
Bug: this will cause no notes to be selected in an edge case where witnesses could have been built.

Consider the most recent subtree that has start height `X`. The wallet has `update_chain_tip(Y)` called, and then the wallet is given blocks to scan in the range `X..=Y`. After that, `update_chain_tip(Z)` is called, but before any more scanning happens, the wallet tries to create a transaction with `X < anchor_height < Y` and `Z - anchor_height < 100`.

It _should_ be possible to generate witnesses in this case, because `Y < anchor_height` is false, but because this conditional is checking for `X <= anchor_height` it fails. I note that the comment below is written assuming that `Y < anchor_height` is being checked.

Reply:
I think that this can be fixed by changing `unscanned.subtree_start_height` to `unscanned.block_range_start`.

Reply:
Yes you came to the same conclusion as we did, which is reassuring.

Reply:
`unscanned.block_range_start` turns out to be not quite correct; the problem is that `MAX(shard_index)` doesn't take into account the anchor height; the anchor could exist in the previous block.